sentry/src/main/java/io/sentry/GsonSerializer.java

```diff
@@ -136,37 +136,14 @@ public GsonSerializer(
     }
   }
 
-  /**
-   * Serialize a SentryEvent to a stream Writer (JSON)
-   *
-   * @param event the SentryEvent
-   * @param writer the Writer
-   * @throws IOException an IOException
-   */
   @Override
-  public void serialize(final @NotNull SentryEvent event, final @NotNull Writer writer)
+  public <T> void serialize(final @NotNull T event, final @NotNull Writer writer)
```

Comment:
Needs some generic name now, like item or entity

```suggestion
  public <T> void serialize(final @NotNull T entity, final @NotNull Writer writer)
```